gprofiler/metadata/metadata_collector.py

```diff
@@ -0,0 +1,35 @@
+import datetime
+import time
+from typing import Dict, Optional, Union
+
+from gprofiler import __version__
+from gprofiler.metadata.cloud_metadata import get_static_cloud_instance_metadata
+from gprofiler.metadata.system_metadata import get_dynamic_system_metadata, get_static_system_info
+
+Metadata = Dict[str, Union[str, int, bool, Dict]]
+
+
+def get_static_metadata(spawn_time: Optional[float]) -> Metadata:
```

Comment:
No point of calculating all of this metadata as it can be costly, instead the static metadata is calculated once on initialization. The dynamic data, like the system uptime is calculated each time the metadata is sent. The metadata is sent on every profile upload, so it would (normally) be sent every 60 seconds